test/middlewares/domain_filter_test.exs

```diff
@@ -16,7 +16,7 @@ defmodule Middlewares.DomainFilterTest do
   test "Filters out requests that do not contain a spider's base_url" do
     middlewares = [Crawly.Middlewares.DomainFilter]
     req = %Crawly.Request{url: "https://www.some_url.com"}
-    state = %{spider_name: :test_spider}
+    state = %{spider_name: :test_spider, crawl_id: "id"}
```

Comment:
testing for whether the crawl_id is set in state by the request storage worker should be in the request storage worker test, not in the middlewares tests.